Handling institutional authors

* papis, version 0.13:

I am having trouble dealing with institutional authors. In order for them to be cited correctly using biblatex they must be exported to bib-files as such: `{{The name of the institutions}}`. Otherwise this gets interpreted as name-surname-pair and results in something like "Name of the institution, T." when referenced.

# how to reproduce the issue
Consider the following bib file `ipcc1990.bib`
```
@report{ipcc1990,
  author={{Intergovernmental Panel on Climate Change}},
  title={Contribution of Working Group I to the First Assessment Report of the Intergovernmental Panel on Climate Change},
  year={1990},
}
```
After importing it via `papis add --from bibtex ipcc1990.bib` it looks something like this:

``` 
author: on Climate Change, Intergovernmental Panel
author_list:
- family: on Climate Change
  given: Intergovernmental Panel
files: []
papis_id: ac0ce27201d0360e806daa3924f77201
ref: ipcc1990
time-added: 2024-07-10-11:23:44
title: Contribution of Working Group I to the First Assessment Report of the Intergovernmental Panel on Climate Change
type: report
year: '1990'

```
The author gets interpreted as name-surname-pair. I could live with that and change it manually to:
```
author: {{Intergovernmental Panel on Climate Change}}
files: []
papis_id: ac0ce27201d0360e806daa3924f77201
ref: ipcc1990
time-added: 2024-07-10-11:23:44
title: Contribution of Working Group I to the First Assessment Report of the Intergovernmental Panel on Climate Change
type: report
year: '1990'
```
Unfortunately, when trying to export that I get an error:

```sh
$ papis export ipcc1990 -f bibtex
[INFO] database.cache: Indexing library. This might take a while...
[ERROR] document: Error reading info file at '~/papis/ipcc1990__ac0ce27201d0360e806daa3924f77201/info.yaml'. Please check it! (Caught exception 'ValueError: while constructing a mapping
  in...'. Use `--log DEBUG` to see traceback)
No document selected!
```

For some reason, this also deletes all metadata from the yaml file except for the id:

```sh
$ cat ~/papis/ipcc1990__ac0ce27201d0360e806daa3924f77201/info.yaml
papis_id: e1f27a34adca7f3764a44fe99b5dcf3a
```
Any idea on how to deal with this? For now, any type of dirty hack would also be much appreciated!

--- a/papis/bibtex.py
+++ b/papis/bibtex.py
@@ -512,6 +512,29 @@ def create_reference(doc: Dict[str, Any], force: bool = False) -> str:
     return ref_cleanup(ref)
 
 
+def author_list_to_author(author_list: List[Dict[str, Any]]) -> str:
+    if not author_list:
+        return ""
+
+    result = []
+    fmt = "{au[family]}, {au[given]}"
+
+    for author in author_list:
+        family = author.get("family")
+        given = author.get("given")
+        if family and given:
+            result.append(fmt.format(au=author))
+        elif family:
+            result.append(f"{{{family}}}")
+        elif given:
+            result.append(f"{{{given}}}")
+        else:
+            # NOTE: empty author, just skip it
+            pass
+
+    return " and ".join(result)
+
+
 def to_bibtex_multiple(documents: List[papis.document.Document]) -> Iterator[str]:
     for doc in documents:
         bib = to_bibtex(doc)
@@ -600,6 +623,8 @@ def to_bibtex(document: papis.document.Document, *, indent: int = 2) -> str:
                 logger.warning(
                     "'journal-key' key '%s' is not present for ref '%s'.",
                     journal_key, document["ref"])
+        elif bib_key == "author" and "author_list" in document:
+            bib_value = author_list_to_author(document["author_list"])
 
         override_key = f"{bib_key}_latex"
         if override_key in document:

--- a/papis/bibtex.py
+++ b/papis/bibtex.py
@@ -394,9 +394,9 @@ def bibtexparser_entry_to_papis(entry: Dict[str, Any]) -> Dict[str, Any]:
             }]),
         _k("author", [{
             "key": "author_list",
-            "action": lambda author: papis.document.split_authors_name([
-                latex_to_unicode(author)
-                ], separator="and")
+            "action": lambda author: (
+                papis.document.split_authors_name([author], separator="and")
+                )
             }]),
     ]
 

--- a/papis/document.py
+++ b/papis/document.py
@@ -254,7 +254,9 @@ def split_author_name(author: str) -> Dict[str, Any]:
     given = " ".join(parts["first"])
     family = " ".join(parts["von"] + parts["last"] + parts["jr"])
 
-    return {"family": family, "given": given}
+    from bibtexparser.latexenc import latex_to_unicode
+
+    return {"family": latex_to_unicode(family), "given": latex_to_unicode(given)}
 
 
 def split_authors_name(authors: Union[str, List[str]],

--- a/tests/test_bibtex.py
+++ b/tests/test_bibtex.py
@@ -122,13 +122,14 @@ def test_to_bibtex_formatting(tmp_config: TemporaryConfiguration) -> None:
     assert to_bibtex(from_data({
         "type": "report",
         "author": "Albert Einstein",
+        "author_list": [{"given": "Albert", "family": "Einstein"}],
         "title": "The Theory of Everything",
         "journal": "Nature",
         "year": 2350,
         "ref": "MyDocument"})
         ) == (
         "@report{MyDocument,\n"
-        "  author = {Albert Einstein},\n"
+        "  author = {Einstein, Albert},\n"
         "  journal = {Nature},\n"
         "  title = {The Theory of Everything},\n"
         "  year = {2350},\n"
@@ -138,10 +139,11 @@ def test_to_bibtex_formatting(tmp_config: TemporaryConfiguration) -> None:
         "type": "misc",
         "ref": "SDbwLashko2019",
         "author": "Alexander Lashkov",
+        "author_list": [{"given": "Alexander", "family": "Lashkov"}],
         "url": "https://github.com/alashkov83/S_Dbw"})
         ) == (
         "@misc{SDbwLashko2019,\n"
-        "  author = {Alexander Lashkov},\n"
+        "  author = {Lashkov, Alexander},\n"
         "  url = {https://github.com/alashkov83/S_Dbw},\n"
         "}")
 
@@ -155,6 +157,7 @@ def test_overridable(tmp_config: TemporaryConfiguration) -> None:
     doc = {
         "type": "report",
         "author": "Albert Einstein",
+        "author_list": [{"given": "Albert", "family": "Einstein"}],
         "title": "Ä α The Theory of Everything & Nothing",
         "title_latex": r"The Theory of Everything \& Nothing",
         "journal": "Nature",
@@ -165,7 +168,7 @@ def test_overridable(tmp_config: TemporaryConfiguration) -> None:
     papis.config.set("bibtex-unicode", True)
     assert to_bibtex(from_data(doc)) == (
         "@report{MyDocument,\n"
-        "  author = {Albert Einstein},\n"
+        "  author = {Einstein, Albert},\n"
         "  journal = {Nature},\n"
         "  title = {The Theory of Everything \\& Nothing},\n"
         "  year = {2350},\n"
@@ -174,7 +177,7 @@ def test_overridable(tmp_config: TemporaryConfiguration) -> None:
     papis.config.set("bibtex-unicode", False)
     assert to_bibtex(from_data(doc)) == (
         "@report{MyDocument,\n"
-        "  author = {Albert Einstein},\n"
+        "  author = {Einstein, Albert},\n"
         "  journal = {Nature},\n"
         "  title = {The Theory of Everything "
         # this will sadly happen, and it makes sense
@@ -195,12 +198,13 @@ def test_ignore_keys(tmp_config: TemporaryConfiguration,
     doc = {
         "type": "report",
         "author": "Albert Einstein",
+        "author_list": [{"given": "Albert", "family": "Einstein"}],
         "year": 2350,
         "ref": "MyDocument"
     }
     assert to_bibtex(from_data(doc)) == (
         "@report{MyDocument,\n"
-        "  author = {Albert Einstein},\n"
+        "  author = {Einstein, Albert},\n"
         "  year = {2350},\n"
         "}")
 
@@ -212,5 +216,28 @@ def test_ignore_keys(tmp_config: TemporaryConfiguration,
 
     assert to_bibtex(from_data(doc)) == (
         "@report{MyDocument,\n"
-        "  author = {Albert Einstein},\n"
+        "  author = {Einstein, Albert},\n"
         "}")
+
+
+def test_import_institution(tmp_config: TemporaryConfiguration) -> None:
+    from papis.bibtex import bibtex_to_dict, to_bibtex
+
+    orig = (
+        "@report{ipcc1990,\n"
+        "  author = {{Intergovernmental Panel on Climate Change}},\n"
+        "  title = {Contribution of Working Group I to the First Assessment "
+        "Report of the Intergovernmental Panel on Climate Change},\n"
+        "  year = {1990},\n"
+        "}"
+    )
+    to_result, = bibtex_to_dict(orig)
+
+    author, = to_result["author_list"]
+    assert not author["given"]
+    assert author["family"] == "Intergovernmental Panel on Climate Change"
+
+    from papis.document import from_data
+
+    from_result = to_bibtex(from_data(to_result))
+    assert from_result == orig

